ci/infra/azure/master-instance.tf

```diff
@@ -16,16 +17,18 @@ resource "azurerm_network_interface_security_group_association" "master" {
   count                     = var.masters
   network_interface_id      = element(azurerm_network_interface.master.*.id, count.index)
   network_security_group_id = azurerm_network_security_group.master.id
+  depends_on                = [azurerm_network_interface.master, azurerm_network_security_group.master, azurerm_linux_virtual_machine.master]
 }
 
 resource "azurerm_linux_virtual_machine" "master" {
   count                 = var.masters
-  name                  = "${var.stack_name}-master-${count.index}-vm"
+  name                  = "${var.stack_name}-master-${count.index}"
   resource_group_name   = azurerm_resource_group.resource_group.name
   location              = azurerm_resource_group.resource_group.location
   zone                  = var.enable_zone ? var.azure_availability_zones[count.index % length(var.azure_availability_zones)] : null
   size                  = var.master_vm_size
   network_interface_ids = [element(azurerm_network_interface.master.*.id, count.index)]
+  depends_on            = [azurerm_network_interface.master]
```

Comment:
Why do you need to mark this dependency in an explicit way? Should we update the existing v4 template too?

Reply:
Seems like there are race condition when `terraform apply/destroy` when use `cloud-init` instead of `virtual_machine_extesion`. If not declared explicitly, terraform script with stuck at some point in apply/destroy process.